### DISCUSSION TITLE
Is there a way to disable dynamic linking for release builds?

### DISCUSSION THREAD

**Discussion by rcorre** [bevyengine/bevy#13510] [Q&A] [answered]
Based on [this SO answer](https://stackoverflow.com/questions/31403430/how-to-switch-dependencies-based-on-build-profile), it is not possible to set profile-specific dependencies.

I'm wondering if I'm missing some other obvious way to disable the `dynamic_linking` feature for release builds. Do folks just edit their `Cargo.toml` every time they want to distribute a build?

Alternatively, is there an obvious way to distribute libbevy with your game? I see `libbevy.so` in the `target/debug` folder, but my game actually links to `libbevy_dylib-db68960f13ec6f9b.so`, which is buried somewhere in `target/debug/deps`.

**Comment by jf908** [comment]
You can enable bevy features using your own feature so I have a feature just called "dev" that has dynamic linking enabled:

```toml
[features]
dev = ["bevy/dynamic_linking"]
```

You have to remember to enable the feature during compilation though. My workflow is to always run `cargo run --features dev` during dev and `cargo run --release` (the feature is disabled by default) for release builds.

Not a perfect solution but doesn't rely on changing Cargo.toml.

**Reply by rcorre** [reply]
I found out you can pass `cargo run --features bevy/dynamic_linking`, so you don't need an intermediate feature, but this still requires you to remember a flag in the normal case, and omit the flag in the release case. I was hoping for a way to default to dynamic with `cargo run`, and omit the flag in release (e.g. `cargo run --no-features bevy/dynamic_linking`, which doesn't seem to exist).

**Reply by jf908** [reply, marked as answer]
Yep, unfortunately cargo just doesn't support that. The only alternatives I can offer is using a wrapper command like [just](https://github.com/casey/just) or personally I have setup an alias in my [config.toml](https://doc.rust-lang.org/cargo/reference/config.html#alias) called dev to make the command a bit shorter: `cargo dev` but of course this does mean using non standard commands that are specific to your project.

**Reply by rcorre** [reply]
Oh that's nice, I like the `alias` idea, thanks!
